lib/trie/destroyer.ex

```diff
@@ -0,0 +1,99 @@
+defmodule MerklePatriciaTree.Trie.Destroyer do
+  @moduledoc """
+  Destroyer is responsible for removing keys from a
+  merkle trie. To remove a key, we need to make a
+  delta to our trie which ends up as the canonical
+  form of the given tree as defined in http://gavwood.com/Paper.pdf.
+
+  Note: this algorithm is non-obvious, and hence why we have a good
+  number of functional and invariant tests. We should add more specific
+  unit tests to this module.
+
+  """
+  alias MerklePatriciaTree.Trie
+  alias MerklePatriciaTree.Trie.Node
+  alias MerklePatriciaTree.ListHelper
+
+  @empty_branch Node.encode_node(:empty, nil)
+
+  @doc """
+  Removes a key from a given trie, if it exists.
+
+  This may radically change the structure of the trie.
+  """
+  def remove_key(trie_node, key, trie) do
+    IO.inspect(["Removing", trie_node, key, trie])
```

Comment:
Left over inspect?

Wanna hookup [Hound](https://houndci.com/) (shameless plug) or maybe have `credo` warnings fail the build?

Reply:
Yeah, can do that. It's only there since this is WIP and it's a bit of a bugger to debug (since the tests don't have tests for intermediate states, so it's kind of hit or miss)